django/db/migrations/operations/models.py

```diff
@@ -716,3 +716,103 @@ def database_backwards(self, app_label, schema_editor, from_state, to_state):
 
     def describe(self):
         return "Change managers on %s" % (self.name, )
+
+
+class CreateIndex(Operation):
+    """
+    Creates an index for a model.
+    """
+
+    def __init__(self, model_name, index):
+        self.model_name = model_name
+        self.index = index
+
+    def deconstruct(self):
+        kwargs = {
+            'model_name': self.model_name,
+            'index': self.index,
+        }
+        if hasattr(self, 'name'):
+            kwargs['name'] = self.name
+        return (
+            self.__class__.__name__,
+            [],
+            kwargs
+        )
+
+    def state_forwards(self, app_label, state):
+        model_state = state.models[app_label, self.model_name.lower()]
+        if not hasattr(self.index, 'model'):
+            self.index.model = state.apps.get_model(app_label, self.model_name)
+        if 'indexes' not in model_state.options:
+            model_state.options['indexes'] = []
+        model_state.options['indexes'].append(self.index)
+
+    def database_forwards(self, app_label, schema_editor, from_state, to_state):
+        to_model = to_state.apps.get_model(app_label, self.model_name)
+        schema_editor.create_index(
+            self.index,
+        )
+
+    def database_backwards(self, app_label, schema_editor, from_state, to_state):
+        from_model = from_state.apps.get_model(app_label, self.model_name)
+        schema_editor.drop_index(
+            self.index,
+        )
+
+    def describe(self):
+        return "Create index on field(s) %s of model %s" % (
+            ', '.join(self.index.fields),
+            self.model_name,
+        )
+
+
+class DropIndex(Operation):
+    """
+    Deletes an index of a model.
+    """
+
+    def __init__(self, model_name, name):
+        self.model_name = model_name
+        self.name = name
+
+    def deconstruct(self):
+        kwargs = {
+            'model_name': self.model_name,
+            'name': self.name
+        }
+        return (
+            self.__class__.__name__,
+            [],
+            kwargs
+        )
+
+    def state_forwards(self, app_label, state):
+        model_state = state.models[app_label, self.model_name.lower()]
+        i, idx = model_state.get_index_by_name(self.name)
+        idx_list = model_state.options['indexes']
+        model_state.options['indexes'] = idx_list[:i] + idx_list[(i + 1):]
```

Comment:
I think you want `options['indexes'].remove(idx_name)` here. That would simplify this code, and I think together with the change in signature mentioned above, remove the need for `get_index_by_name` as well.

[List.remove](https://docs.python.org/3/tutorial/datastructures.html#more-on-lists)